src/Symfony/Component/Security/Guard/Authenticator/GuardAuthenticatorInterface.php

```diff
@@ -0,0 +1,168 @@
+     * @param UserProviderInterface $userProvider
+     *
+     * @throws AuthenticationException
+     *
+     * @return UserInterface|null
+     */
+    public function getUser($credentials, UserProviderInterface $userProvider);
+
+    /**
+     * Returns true if the credentials are valid.
+     *
+     * If any value other than true is returned, authentication will
+     * fail. You may also throw an AuthenticationException if you wish
+     * to cause authentication to fail.
+     *
+     * The *credentials* are the return value from getCredentials()
+     *
+     * @param mixed         $credentials
+     * @param UserInterface $user
+     *
+     * @throws AuthenticationException
+     */
+    public function checkCredentials($credentials, UserInterface $user);
+
+    /**
+     * Create an authenticated token for the given user.
+     *
+     * If you don't care about which token class is used or don't really
+     * understand what a "token" is, you can skip this method by extending
+     * the AbstractGuardAuthenticator class from your authenticator.
+     *
+     * @see AbstractGuardAuthenticator
+     *
+     * @param UserInterface $user
+     * @param string        $providerKey The provider (i.e. firewall) key
+     *
+     * @return GuardTokenInterface
+     */
+    public function createAuthenticatedToken(UserInterface $user, $providerKey);
+
+    /**
+     * Called when authentication executed, but failed (e.g. wrong username password).
+     *
+     * This should return the Response sent back to the user, like a
+     * RedirectResponse to the login page or a 403 response.
+     *
+     * If you return null, the request will continue, but the user will
+     * not be authenticated. This is probably not what you want to do.
+     *
+     * @param Request                 $request
+     * @param AuthenticationException $exception
+     *
+     * @return Response|null
+     */
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception);
+
+    /**
+     * Called when authentication executed and was successful!
+     *
+     * This should return the Response sent back to the user, like a
+     * RedirectResponse to the last page they visited.
+     *
+     * If you return null, the current request will continue, and the user
+     * will be authenticated. This makes sense, for example, with an API.
+     *
+     * @param Request        $request
+     * @param TokenInterface $token
+     * @param string         $providerKey The provider (i.e. firewall) key
+     *
+     * @return Response|null
+     */
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, $providerKey);
+
+    /**
+     * Does the authenticator supports the given Request ?
+     *
+     * If it does, authentication based on the given request will continue.
+     * If not the next (if any) authenticator will be requested
+     * until at one supports the current request.
```

Comment:
>
If this returns true, authentication will continue (e.g. getCredentials() will be called).
If false, this authenticator is done. The next (if any) authenticators will be called and
may authenticate the user, or leave the user as anonymous.